src/org/javarosa/xpath/expr/XPathPathExpr.java

```diff
@@ -355,16 +285,16 @@ public String toString () {
         return sb.toString();
     }
 
-    public boolean equals (Object o) {
+    public boolean equals(Object o) {
         if (o instanceof XPathPathExpr) {
-            XPathPathExpr x = (XPathPathExpr)o;
+            XPathPathExpr x = (XPathPathExpr) o;
 
             //Shortcuts for easily comparable values
-            if(init_context != x.init_context || steps.length != x.steps.length) {
+            if (init_context != x.init_context || steps.length != x.steps.length) {
                 return false;
             }
 
-            return ExtUtil.arrayEquals(steps, x.steps) && (init_context == INIT_CONTEXT_EXPR ? filtExpr.equals(x.filtExpr) : true);
+            return ExtUtil.arrayEquals(steps, x.steps) && (init_context != INIT_CONTEXT_EXPR || filtExpr.equals(x.filtExpr));
```

Comment:
I might have to wait for tomorrow to be fresh again to really be sure that this does the same, but I'm guessing you've just let IntelliJ simplify the boolean expression :+1:

Reply:
Yes.